smtp-mail.cpp

```diff
@@ -50,14 +52,48 @@ char *getCurrTime()
 	return buffer;
 }
 
+std::vector<std::string> stringTokenize(std::string searchString ,std::regex searchPattern)
+{
+	std::vector<std::string> vec;
+	auto token_start = std::sregex_iterator(searchString.begin(), searchString.end(), searchPattern);
+	auto token_end = std::sregex_iterator();
+	for (std::sregex_iterator i = token_start; i != token_end; ++i)
+	{
+		std::smatch match = *i;
+		std::string matchValue = match.str();
+		vec.emplace_back(matchValue);
+	}
+	return vec;
+
+}
+
 void compose_payload(vector<std::string>* &payload, const EmailCfg *emailCfg, const char* msg)
 {
 	payload->push_back("Date: " + string(getCurrTime()) + "\r\n");
 	payload->push_back("To: " + emailCfg->email_to_name + " <" + emailCfg->email_to + "> \r\n");
+	
+	// Parse address and name to compose CC pairs for payload
+	std::regex searhPattern("[^\\,]+");
 	if (!emailCfg->email_cc.empty())
-		payload->push_back("CC: " + emailCfg->email_cc_name + " <" + emailCfg->email_cc + "> \r\n");
+	{
+		std::vector<string> ccAddress =  stringTokenize(emailCfg->email_cc,searhPattern);
+		std::vector<string> ccNames =  stringTokenize(emailCfg->email_cc_name,searhPattern);
```

Comment:
Please compare that number of Cc addresses and Cc names is the same. Also check similarly for Bcc names and addresses somewhere even though they are not put into payload.